### ISSUE TITLE
Secret search path joined into one path instead of iteration over paths

### ISSUE THREAD

**Issue by kokovikhinkv** [petems/petems-hiera_vault#29]
```
hierarchy:
  - name: "Hiera-vault lookup"
    lookup_key: hiera_vault
    options:
      confine_to_keys:
        - '^vault_.*'
      ssl_verify: false
      address: 'http://127.0.0.1:8200'
      token: '/etc/puppetlabs/puppetserver/vault_token'
      default_field: value
      mounts:
        puppet:
          - "%{::trusted.certname}"
          - common
```

Output:
```
  Environment Data Provider (hiera configuration version 5)
    Using configuration "/etc/puppetlabs/code/environments/env_name/hiera.yaml"
    Hierarchy entry "Hiera-vault lookup"
      No such key: "vault_key"
      [hiera-vault] Client configured to connect to http://127.0.0.1:8200
      [hiera-vault] Looking in path puppet/certname/common/vault_key
```

Expecter behavior:
```
[hiera-vault] Looking in path puppet/certname/vault_key
[hiera-vault] Looking in path puppet/common/vault_key
```

**Comment by petems** [comment]
This should be fixed by #31, I'm going to cut a release and can you test and confirm?

**Comment by petems** [comment]
Closing due to lack of response, feel free to re-open if the latest release doesnt work for you! 😄
